### DISCUSSION TITLE
Webfollower w Libre3, Signal Missed

### DISCUSSION THREAD

**Discussion by Havanna22** [NightscoutFoundation/xDrip#2888] [Q&A] [answered]
I have used Xdrip+ with Webfollower and Libre3 for months now but now it suddenly says Signal Missed.
This coincides with an update of LibreLinkUp, maybe there is a connection?
Libre3 app works correctly, LibreLinkUp App works correctly showing the glucose level, jut Xdrip+ gives Signal Missed
I have an Android version 13.
I have tried both with the Latest release XDrip+ and also the latest pre-release and restarted phone.
What else can I try to get it to work?

**Comment by rockygaur** [comment]
This seems to be related to the new LibreLinkUp release 4.7.0 which has just gone live. I’ve updated my LibreLinkUp app and it is working, but still not getting reading on Xdrip+.


![513AEC98-8CD4-4145-A581-D94DF0287831](https://github.com/NightscoutFoundation/xDrip/assets/97365523/45fb8f63-d70a-428b-bf05-5e0cd7d09d5f)

**Reply by Havanna22** [reply]
Checked my log and I have same error code and then (now) Process failure
![Screenshot_20230616_023240_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/136760557/28df3d6b-7747-4163-996d-0d6348aaa514)

**Reply by hpcolus** [reply]
Part of the issue is that the webfollower depends on data from librelinkup.  Since yesterday  abbot broke the code.  Librelinkup is now lagging 20min-data (before was almost on realtime)  Please call abbot for technical support because is not acceptable to have a follower with the 20-min lagging of glucose data even in their application (check your librelinkup app and confirm the lagging).  The more people call technical support the better.  However, even with the latency of 20min, xDrip should check what is going on.  BTW, I too confirm the issue

**Reply by Havanna22** [reply]
I don't have a latency in XDrip, I have Signal Missed, no readings at all since yesterday
And no latency that I can see in LibreLinkUp, no point calling Abbott for me, their SW work on my phone

**Reply by dxnxz** [reply]
![Screenshot_20230616_135117_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/60526124/d19241b0-b69e-4d60-952f-434af0ec7be0)

same here

**Reply by FloSchl8** [reply]
> Checked my log and I have same error code and then (now) Process failure
> 
> ![Screenshot_20230616_023240_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/136760557/28df3d6b-7747-4163-996d-0d6348aaa514)
> 
> 
> 
> 

The entry from 00:04 says you have to reaccept the Libre Terms of Service (see the 'steps' part). Had the same issue while debugging the API.

Unfortunately in the LibeLinkUp App it won't display the ToS again and I had to logout and login again to accept them.

**Comment by Navid200** [comment]
I know that you guys are frustrated.
We can see this discussion.  Opening an issue is not going to speed things up.

We first need to know if we can fix this or not.  In other words, can we change something in xDrip to address this?
The answer may be yes or no.

If yes, someone may already assign it to themselves and fix it with no need to opening an issue.
If we can fix it, but no one is going to fix it right away, or if we know xDrip is the cause of the problem, but, we have no idea how, we may need to open an issue.

We don't open an issue as a first step and take it from there.

Along those lines, can you explain why the Companion app mode cannot be used to transfer readings to xDrip?
I'm not saying that is possible.  I am asking if it is.
If a workaround is available, it is good to know.  If there is no workaround, it is good to clarify.

**Reply by hpcolus** [reply]
It does not work for Freestyle Libre 3.  According to xDrip+ user's manual companion app only works for G6 (master only), One G7 (master only), CamAPS, Guardian, and MiniMed.  As of 1:30 EST it appears that librelinkup now sinchonizes on real time. but still xDrip shows errors on reading.

**Reply by nochrix** [reply]
There is an analysis of what has to be changed to get data from the API again: https://github.com/timoschlueter/nightscout-librelink-up/pull/101
I think this might help for xDrip too.

**Reply by robbless** [reply]
I can confirm also that the Web Follower is not working anymore for LibreLinkUp. I have logged out of LibreLinkUp and logged in again and then needed to accept the terms and conditions again. This did not solve the problem, but now the error logs are different (https://photos.app.goo.gl/nd7ndaSYa1GvCJVi8).

I'm not a developer but now it looks like it fails because the web follower is not using the new Version - 4.7.0

**Reply by HoLi1969** [reply]
Hello, I confirm the status of robbless. I proceeded the same way and have some result. Issue seems clearly linked to the new LibreLinkUp version. Hope our brilliant guys here will find a patch for x-Drip.

**Reply by schorschl** [reply]
I can also confirm that the API change broke the "Web Follower" functionality for Abbott Freestyle Libre 3 (and probably all LLU users). It does not work any more since yesterday evening. Please check Timo Schlueter's solution. Thanks for the great work, I love xDrip+ and it gives so much insight, guidance and provides tools to get the management done perfectly with Pens.

**Reply by HoLi1969** [reply]
Where can I find this?

**Reply by schorschl** [reply]
You can't as a user. The code of the xDrip+ app needs to be modified by the developers to be able to "talk" to the Libre API, which changed the way it communicates with Abbott's own Libre Link Up App. At least that's my interpretation of the situation. :-)

**Reply by HoLi1969** [reply]
Probably I misunderstood. Is there already a solution to get X-Drip back to Web Follower?

**Reply by HoLi1969** [reply]
Ok, thanks

**Comment by dreimer1986** [comment]
![Screenshot_20230616-121455](https://github.com/NightscoutFoundation/xDrip/assets/29841368/f57c5b5d-b5a7-4fd8-a9df-940d390319f2)
I got this warning by mail a few days ago...

**Reply by HoLi1969** [reply]
That's what we speak about. Last night the App asked to be updated or did automatically to new version. Since then X-Drip is offline.

**Comment by dxnxz** [comment]
I just installed linkupconnect as a workaround. works fine for me so far.

**Reply by dxnxz** [reply]
https://github.com/cmtjk/LinkUpConnect/releases

**Comment by Havanna22** [comment]
Does LinkUpConnect work with Libre 3? If so, what's the trick to get it to
work? It refers to Libre2 when trying to activate connection to XDrip+
Which source in XDrip+ should I select if I configure it manually?
I get the attached error message.
![Screenshot_20230616_143042_Gallery](https://github.com/NightscoutFoundation/xDrip/assets/136760557/5782a240-5eca-4623-960d-080d994591f5)


Den fre 16 juni 2023 14:05dxnxz ***@***.***> skrev:

**Reply by dxnxz** [reply]
yes! you have to fill in your Libre Link Up credentials in linkupconnect. in xdrip you have to select Libre (patched) as data source (see screenshot). then it should work.
![Screenshot_20230616_144828_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/60526124/82057ae8-64fd-4a1e-acdb-f872bbdbfcea)


I allowed linkupconnect to keep showing me notifications because I have the feeling that otherwise it gets stopped by Android. might be some battery optimization feature or so.
![Screenshot_20230616_144957_LinkUpConnect](https://github.com/NightscoutFoundation/xDrip/assets/60526124/67290533-ef2b-4b09-bcd0-7fb975677671)

hope that helps!

**Reply by nochrix** [reply]
Maybe you have to change the URL of the server, depending on your region. api-de.libreview.io seems the be the german service. Does it work, if you change it to api-eu.libreview.io for example? 

Here it says that possible values for the region are AE, AP, AU, CA, DE, EU, EU2, FR, JP, US :
https://github.com/timoschlueter/nightscout-librelink-up/blob/main/README.md

**Reply by jornh71** [reply]
For Norway it works with EU.

**Comment by Navid200** [comment]
Very well.
I have reopened the issue.

**Comment by dxnxz** [comment, marked as answer]
> Maybe you have to change the URL of the server, depending on your region. api-de.libreview.io seems the be the german service. Does it work, if you change it to api-eu.libreview.io for example?
> 
> Here it says that possible values for the region are AE, AP, AU, CA, DE, EU2, EU2, FR, JP, US : https://github.com/timoschlueter/nightscout-librelink-up/blob/main/README.md

Thanks for the hint! Haven't thought about that.

**Reply by andrenicGH** [reply]
FYI, I'm getting the same error as Havanna22 (see below).

I'm a Norwegian Libre 3 user.
Perhaps the linkupconnext app also lets me forward 1min resolution data to xDrip+, rather than the 5min currently possible when using WebFollower as data source?

![Screenshot_20230616_143042_Gallery](https://github.com/NightscoutFoundation/xDrip/assets/136760557/5782a240-5eca-4623-960d-080d994591f5)

**Reply by dxnxz** [reply]
i guess you'd have to put api-eu.libreview.io as @nochrix stated. "api-de" is probably the german server.

**Reply by mpvbasilio** [reply]
indeed! changed mine and it works.

**Reply by andrenicGH** [reply]
Here also, did not realize this was editable for me until now, sorry about that :)

**Reply by HoLi1969** [reply]
Many thanks to all the brilliant guys here. The app works also for us. Perfect. Would be great to be informed when this will be implemented into x-Drip like the old llu app.

**Comment by Havanna22** [comment]
Yes now it works, with eu! I get measurements in XDrip+ and now also on my Galaxy watch again.
Thank you a million times!

**Comment by jeveli** [comment]
Yes this works. Got measurements in xdrip and on my galaxy watch

**Comment by Navid200** [comment]
This is now resolved: https://navid200.github.io/xDrip/docs/Libre3_Follow.html

**Reply by schorschl** [reply]
I can confirm that it is working again after the reset. I am using the latest nightly build. d46b342-2023.06.17

**Reply by mpvbasilio** [reply]
I second this. Many thanks!!!

**Comment by paolomonica** [comment]
I've  a smartwatch galaxy pro 5. Now xdrip receive the data but the smartwatch any. Why?
